location/utils.py

```diff
@@ -99,3 +99,81 @@ def compute_total_gyration(data, sr_col='stay_region'):
         temp_sum += loc_cnt[loc] * vincenty(geohash.decode(loc), r_cm).m ** 2
 
     return math.sqrt(1/N * temp_sum)
+
+
+def compute_gyration(data,
+                     sr_col='stay_region',
+                     k=None,
+                     remove_dist_pt=True,
+                     home=None,
+                     dist_pt_th=50000):
+    """
+    Compute radius of gyration.
+
+    Parameters:
+    -----------
+
+    data: DataFrame
+        Location data.
+
+    sr_col: str
+        Column name for stay region.
+        Default is 'stay_region'.
+
+    k: int
+        k-th radius of gyration.
+        Default is None, in this case return total radius of gyration.
+        k-th radius of gyration is the radius gyration compuated up to
+        the k-th most frequent visited locations.
+
+    remove_dist_pt: bool
```

Comment:
I'd recommend filtering points at the beginning of the analysis. This sort of piece-wise filtering makes it i) harder to explain to others and ii) easier to make mistakes. You should explain why we are filtering here. And, if you decide to keep it, I'd recommend making the default value False here so that we are explicit about filtering.